ttnn/cpp/ttnn/tensor/tensor.cpp

```diff
@@ -91,6 +133,86 @@ Tensor::Tensor(const Storage storage, const ttnn::Shape shape, DataType dtype, L
     this->tensor_attributes->metadata_populated = true;
 }
 
+Tensor::Tensor(
+    const std::vector<Device *>& workers,
+    uint32_t num_buffers,
+    std::optional<DistributedTensorConfig> distributed_tensor_config) :
+    tensor_id(std::nullopt),
+    tensor_attributes(std::make_shared<TensorAttributes>()),
+    workers(workers),
+    deallocate_through_destructor(false) {
+    // When creating a device tensor, specify workers.
+    // When creating a host tensor, specify num_buffers.
```

Comment:
You probably want two different constructors. One for vector<Device> and another for num_buffers.